src/main.ts

```diff
@@ -586,34 +586,33 @@ export default class SRPlugin extends Plugin {
         new Notice(t("ALL_CAUGHT_UP"));
     }
 
-    findDeckPath(note: TFile): string[] {
-        let deckPath: string[] = [];
+    findAllDeckPaths(note: TFile): string[][] {
+        let deckPaths: string[][] = [];
         if (this.data.settings.convertFoldersToDecks) {
-            deckPath = note.path.split("/");
-            deckPath.pop(); // remove filename
-            if (deckPath.length === 0) {
-                deckPath = ["/"];
+            deckPaths[0] = note.path.split("/");
+            deckPaths.pop(); // remove filename
+            if (deckPaths.length === 0) {
+                deckPaths[0] = ["/"];
             }
         } else {
             const fileCachedData = this.app.metadataCache.getFileCache(note) || {};
-            const tags = getAllTags(fileCachedData) || [];
+            const tagsInThisNote = getAllTags(fileCachedData) || [];
 
-            outer: for (const tagToReview of this.data.settings.flashcardTags) {
-                for (const tag of tags) {
+            for (const tagToReview of this.data.settings.flashcardTags) {
+                for (const tag of tagsInThisNote) {
                     if (tag === tagToReview || tag.startsWith(tagToReview + "/")) {
-                        deckPath = tag.substring(1).split("/");
-                        break outer;
+                        deckPaths.push(tag.substring(1).split("/"));
```

Comment:
This is the crux of the change. Just append to an array here instead of early-returning from the loop.